### PR TITLE
Docker, environmental variables and CI

### PR DESCRIPTION
**Pull request by Chasbob**
# Environmental Variables
If config.json is not found then the environment will be checked for the respective variables

---
# Docker

* 'Fat' docker image
* Compose which can also be used to easily inject the config through environment variables

---

# CI
Github actions for:
* Python
* Docker

---

# Suggestions

The project could be linked to docker hub like [this](https://hub.docker.com/r/chasbob/food-flex) which would automatically create a publicly accessible image (without containing any secrets)

---

### CODE REVIEW

**Comment by danielfspencer** [comment]
Awesome work :tada: 
Have you tested the configuring from environment variables? It's just because some of the values have to be ints and others have to be strings:
```python
{
  "token": str,
  "server_id": int,
  "command_prefix":str,
  "admin_ids": [ints],
  "main_channel_id": int,
  "leaderboard_channel_id": int
}
```
Does python automatically determine the type from quotes (or the lack of) on environment vars? Not experienced in that myself.

**Comment by Chasbob** [comment]
I believe everything is running correctly. I will double check that python is getting the right types.

Here's a sample output when running with creds for the test server.
```
09:27:11  INFO      config: Loading config...
09:27:11  WARN      config: ↳ Cannot open config/config.json
09:27:11 DEBUG      config: ↳ Config loaded from environment
09:27:11 DEBUG         bot: Creating bot (cmd prefix '!')...
09:27:11  INFO    foodflex: Starting Food Flex v2.0.1
09:27:11 DEBUG      static: Loading quotes & strings...
09:27:11  INFO      static: ↳ Quotes & strings loaded
09:27:11 DEBUG        data: Loading state...
09:27:11  INFO        data: ↳ No existing state file, starting with blank state
09:27:11 DEBUG        data: Saving state...
09:27:11  INFO        data: State saved
09:27:11 DEBUG        data: Loading leaderboard...
09:27:11  INFO        data: ↳ No existing leaderboard file, starting with blank
09:27:11 DEBUG        data: Saving leaderboard...
09:27:11  INFO        data: Leaderboard saved
09:27:11  INFO    foodflex: period='submissions', mode='automatic' people=0
09:27:11  INFO    foodflex: Starting bot...
09:27:14 DEBUG         bot: Finding text channels...
09:27:14  INFO         bot: Food Flex is online!
```

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+volumes:
+    food-flex-data:
+
+services:
+    food-flex:
+        container_name: 'food-flex'
+        build: .
+        env_file: .env
+        image: chasbob/food-flex
```
Comment (Chasbob) [inline docker-compose.yml]: This would need to be updated to reflect the correct name.

File: foodflex/foodflex.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 def main():
     logger.info(f'Starting Food Flex v{__version__}')
     static.load()
+    data.set_paths(config.data_root)
```
Comment (Chasbob) [inline foodflex/foodflex.py]: This call was added instead of `data` internally referencing `config` as it may not have been loaded.

**Comment by danielfspencer** [comment]
Looks good! Are there any remaining issues that you are aware of?

**Comment by wrussell1999** [comment]
@Chasbob  any problems or ready to merge? :D

**Comment by Chasbob** [comment]
I'm happy with this, I'd just suggest you have a look over the README and see if you're happy with it.

**Pull request merged**